x-pack/plugins/security_solution/public/management/pages/trusted_apps/service/validate_trusted_app_http_request_body.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { HttpStart } from 'kibana/public';
+import {
+  PostTrustedAppCreateRequest,
+  PutTrustedAppUpdateRequest,
+} from '../../../../../common/endpoint/types';
+import { HttpRequestValidationError } from './errors';
+import { sendGetAgentPolicyList } from '../../policy/store/services/ingest';
+import { AGENT_POLICY_SAVED_OBJECT_TYPE } from '../../../../../../fleet/common';
+
+/**
+ * Validates that the Trusted App is valid for sending to the API (`POST` + 'PUT')
+ *
+ * @throws
+ */
+export const validateTrustedAppHttpRequestBody = async (
+  http: HttpStart,
+  trustedApp: PostTrustedAppCreateRequest | PutTrustedAppUpdateRequest
+): Promise<true> => {
+  const failedValidations: string[] = [];
+
+  // Validate that the Policy IDs are still valid
```

Comment:
After looking through the server code, I found that the only validation we need is for Policy IDs. The UI already validates that the combination of `OS<>Entries` meets the expected combinations, so I did not implement that (the server side Schema) here.